File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        start, end = 0, len(nums) - 1
        while start < end:
            center = (start + end) // 2
            if nums[center] == nums[center + 1] and center % 2 == 0:
                start = center + 1
            elif nums[center] == nums[center - 1] and center % 2 == 0:
                end = center - 1
            elif nums[center] == nums[center + 1] and center % 2 == 1:
                end = center - 1
            elif nums[center] == nums[center - 1] and center % 2 == 1:
                start = center + 1
            else:
                return nums[center]
        return nums[start]

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int start = 0;
        int end = nums.length - 1;
        while (start < end){
            int center = (start + end) / 2;
            if(center == 0 || center == nums.length - 1)
                return nums[center];
            if(nums[center] == nums[center + 1] && center % 2 == 0)
                start = center + 1;
            else if(nums[center] == nums[center - 1] && center % 2 == 0)
                end = center - 1;
            else if(nums[center] == nums[center + 1] && center % 2 == 1)
                end = center - 1;
            else if(nums[center] == nums[center - 1] && center % 2 == 1)
                start = center + 1;
            else
                return nums[center];
            }
        return nums[start];
    }
}